components/authz-service/engine/opa/opa_internal_test.go

```diff
@@ -549,3 +429,864 @@ func randomPolicies(i int) []map[string]interface{} {
+
+			statements := make(map[string]interface{}, statementCount)
+			for k := 0; k < statementCount; k++ {
+				stID := fmt.Sprintf("statement-%v", k)
+
+				var statementActions []string
+				var statementRole string
+
+				// determine if the statement has actions or a role based on a coin toss
+				coinToss := rand.Intn(2)
+				if coinToss == 1 {
+					// no actions, just a role
+					statementActions = nil
+
+					statementRole = getRandomRole(roleMap)
+				} else {
+					// no role, just a list of actions
+					statementRole = ""
+
+					statementActionCount := rand.Intn(len(actions))
+					// the statement must have at least one action
+					if statementActionCount == 0 {
+						// so we use the 0 case as All Actions
+						statementActions = []string{"*"}
+					} else {
+						rand.Shuffle(len(actions), func(x, y int) { actions[x], actions[y] = actions[y], actions[x] })
+						statementActions = actions[:statementActionCount]
+					}
+				}
+
+				var statementProjects []string
+				projectCount := rand.Intn(len(allProjects))
+				// there can never be 0 projects in a statement
+				if projectCount == 0 {
+					// so we use 0 as the All Projects case
+					statementProjects = []string{"~~ALL-PROJECTS~~"}
+				} else {
+					rand.Shuffle(len(allProjects), func(x, y int) { allProjects[x], allProjects[y] = allProjects[y], allProjects[x] })
+					statementProjects = allProjects[:projectCount]
+				}
+
+				statements[stID] = map[string]interface{}{
+					"actions":   statementActions,
+					"role":      statementRole,
+					"resources": "*",     // v2.1 custom policies can only have "*" resources
+					"effect":    "allow", // TODO write separate test setup with deny cases
+					"projects":  statementProjects,
+				}
+			}
+
+			// generate btwn range 1..total number of members
+			memberCount := rand.Intn((len(members) - 1) + 1)
+
+			// shuffle the array of possible members,
+			// then we take a slice from 0 to the random memberCount to get a randomized member list for the policy
+			rand.Shuffle(len(members), func(m, n int) { members[m], members[n] = members[n], members[m] })
+
+			policyMap[id] = map[string]interface{}{
+				"members":    members[:memberCount],
+				"statements": statements,
+			}
+		}
+	}
+
+	return policyMap, roleMap
+}
+
+func v2BaselinePoliciesAndRoles() (policies map[string]interface{}, roles map[string]interface{}) {
+	// this file includes system, migrated legacy, and chef-managed policies
+	// and chef-managed roles
+	jsonFile, err := os.Open("example_v2/real_world_2p1_store.json")
+	if err != nil {
+		fmt.Println(err)
+	}
+	defer jsonFile.Close()
+	byteValue, _ := ioutil.ReadAll(jsonFile)
+	var pr struct {
+		Policies map[string]interface{} `json:"policies"`
+		Roles    map[string]interface{} `json:"roles"`
+	}
+	json.Unmarshal(byteValue, &pr)
+
+	return pr.Policies, pr.Roles
+}
+
+func v2BaselineAndProjectPolicies(count int) (policies map[string]interface{}, projects []string) {
```

Comment:
this helper returns all baseline policies + 3 policies for each project, to reflect our current behavior of generating `owner, editor, viewer` policies for each new project.